src/celeritas/decay/interactor/MuDecayInteractor.hh

```diff
@@ -0,0 +1,242 @@
+    Secondary* secondaries = allocate_(1);
+    if (secondaries == nullptr)
+    {
+        // Failed to allocate secondaries
+        return Interaction::from_failure();
+    }
+
+    real_type electron_energy_frac{};
+    real_type electron_nu_energy_frac{};
+    do
+    {
+        do
+        {
+            electron_nu_energy_frac = sample_electron_nu_(rng);
+        } while (generate_canonical(rng)
+                 > electron_nu_energy_frac
+                       * (real_type{1} - electron_nu_energy_frac));
+
+        electron_energy_frac = generate_canonical(rng);
+    } while (electron_nu_energy_frac < real_type{1} - electron_energy_frac);
+
+    // Momentum of secondaries at rest frame
+    auto charged_lep_energy
+        = this->calc_momentum(electron_energy_frac, shared_.electron_mass);
+
+    // Apply a spherically uniform rotation to the decay
+    IsotropicDistribution rotate;
+    Real3 charged_lep_dir = rotate(rng);
+
+    // Boost secondaries to the lab frame
+    auto charged_lep_4vec = this->to_lab_frame(
+        charged_lep_dir, charged_lep_energy, shared_.electron_mass);
+
+    // Return electron only
+    Interaction result = Interaction::from_decay();
+    result.secondaries = {secondaries, 1};
+    result.secondaries[0].particle_id = is_muon_ ? shared_.ids.electron
+                                                 : shared_.ids.positron;
+    // Interaction stores kinetic energy; FourVector stores total energy
+    result.secondaries[0].energy
+        = Energy{charged_lep_4vec.energy - shared_.electron_mass.value()};
+    result.secondaries[0].direction = make_unit_vector(charged_lep_4vec.mom);
+
+    return result;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Boost secondary to the lab frame.
+ *
+ * \note This assumes the primary to be at rest and, thus, there is no need
+ * to perform an inverse boost of the primary at the CM frame.
+ */
+CELER_FUNCTION FourVector MuDecayInteractor::to_lab_frame(Real3 const& dir,
+                                                          MevMomentum momentum,
+                                                          Mass mass) const
+{
+    CELER_EXPECT(is_soft_unit_vector(dir));
+    CELER_EXPECT(momentum > zero_quantity());
+    CELER_EXPECT(mass >= zero_quantity());
+
+    Real3 p = dir * momentum.value();
+    FourVector lepton_4vec{
+        p, std::sqrt(ipow<2>(norm(p)) + ipow<2>(mass.value()))};
+    boost(boost_vector(inc_fourvec_), &lepton_4vec);
+
+    return lepton_4vec;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Calculate final particle momentum (or kinetic energy) from its sampled
+ * fractional energy.
+ */
+CELER_FUNCTION units::MevMomentum
+MuDecayInteractor::calc_momentum(real_type energy_frac, Mass mass) const
+{
+    return MevMomentum{
+        std::sqrt(ipow<2>(energy_frac * max_energy_)
+                  + 2 * energy_frac * max_energy_ * mass.value())};
```

Comment:
Maybe this function should just be electron momentum? Not passing in the mass (just used `shared_.electron_mass`) and instead passing in `Energy{energy_frac * max_energy_}`.

Reply:
This was originally as is because it was the same function for all three secondaries. I kept it isolated for future refactoring if we need to add the neutrinos, but it could technically just be in the main interactor body, since it is just used once.